aten/src/ATen/native/mps/OperationUtils.mm

```diff
@@ -310,42 +310,43 @@ void printTensorNDArray(const Tensor& t) {
         };
 
         runMPSGraph(stream, cachedGraph->graph(), feeds, results);
-#if _DEBUG
-        NSLog(@"%@", [cachedGraph->graph() debugDescription]);
-        TORCH_WARN("We have a non-contiguous tensor in copy_from_mps with key ", key);
-
-        //// Update the Blit sourceBuffer to the result of this operation
-        printTensorNDArray(output);
-#endif
         return resultBuffer;
       }
     }
   }
   return nil;
 }
 
-
-
-Placeholder::Placeholder(MPSGraphTensor* mpsGraphTensor, const Tensor& self, MPSShape *mpsShape)
+Placeholder::Placeholder(MPSGraphTensor* mpsGraphTensor, const Tensor& src,
+                         MPSShape *mpsShape, bool check_view)
 {
-  TORCH_CHECK(self.is_mps(), "Placeholder storage has not been allocated on MPS device!");
-  // extract the pointer to MTLBuffer from the Tensor's storage
-  id<MTLBuffer> selfBuf = __builtin_bit_cast(id<MTLBuffer>, self.storage().data());
-  const size_t buf_size = [selfBuf length];
+  Tensor src_ = src;
+  TORCH_CHECK(src_.is_mps(), "Placeholder storage has not been allocated on MPS device!");
+    // extract the pointer to MTLBuffer from the Tensor's storage
+  id<MTLBuffer> srcBuf = __builtin_bit_cast(id<MTLBuffer>, src.storage().data());
+  if (!src.is_contiguous()) {
+    id<MTLBuffer> gatherTensor = gatherViewTensor(src, srcBuf);
+    if (gatherTensor) {
+      srcBuf = gatherTensor;
+    } else {
+      src_ = src.contiguous();
```

Comment:
Should this be srcBuf?
Not sure why `srcBuf` or `src_` is set here since both are used later.

Reply:
Ah srcBuf is the underlying data ptr `MTLBuffer` whereas src is the Tensor. The idea is that before we need to materialize the memory from the previous view operation (gather), we update the underlying ptr to right Metal buffer.